presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -250,6 +251,12 @@ public Session beginTransactionId(TransactionId transactionId, TransactionManage
             if (catalogProperties.isEmpty()) {
                 continue;
             }
+            try {
+                accessControl.checkCanAccessCatalog(transactionId, getIdentity(), catalogName);
+            }
+            catch (AccessDeniedException ignore) {
+                throw new PrestoException(NOT_FOUND, "Catalog does not exist: " + catalogName);
```

Comment:
This is confusing. Is this an `AccessDeniedException` or a catalog not found problem?